youtube_dl/extractor/dumpert.py

```diff
@@ -50,6 +69,7 @@ def _real_extract(self, url):
                 'url': uri,
```

Comment:
Is the issue that the `uri` is a manifest (m3u8)? Or is it actually a direct mp4 link with a confusing format?

If the former, the typical pattern for handling format URLs is
```py
            ext = determine_ext(uri)
            if ext == 'm3u8':
                formats.extend(self._extract_m3u8_formats(uri, ...)
            elif ext == ...: # other manifests, eg mpd
                ...
            else:
               formats.append({
                   'url': uri,
                   ...
               })
```
This is so common that it should have been abstracted in a common routine, I think, but perhaps one with too many arguments. Anyhow it automatically extracts all the available formats from the manifest so that they can be listed. If appropriate, do the same here.

Reply:
I have to admit I did see that going on in some other extractors but thought it wasn't meant to be used for non-master playlists (which is what they are here). Looking at the code, I saw no way to add in the quality argument through `_extract_m3u8_formats` so I did that in a separate loop afterwards.